### DISCUSSION TITLE
importing from JSON + object refs

### DISCUSSION THREAD

**Discussion by eriqjaffe** [fabricjs/fabric.js#7772] [Q&A] [answered]
So I have a project with some grouped objects where I'm trying to implement some save & load functionality.  Thing is, objects in groups render properly after reloading them from JSON but then I can't seem to touch them after that.  I set up an example fiddle:

https://jsfiddle.net/eriqjaffe/jo25d8qs/8/

Notice that at first, the "Background", "Square" and "Line" buttons will all randomly change the fill color of the respective object.  But after you reload from JSON (with the first button), the "Line" button no longer works.  I get the feeling that it's something I'm doing wrong (or not doing at all) with the reviver but I'm not sure what it is.  Any help would be appreciated.

**Comment by ShaMan123** [comment, marked as answer]
You load from json and then change the fill of the line of the group you've created (`ElementGroup`) that isn't part of canvas because you've loaded other objects from json instead. So it changed the fill but on an object you didn't mean to.

It's an oversight not regarding fabric. In other words you set a global var that should be local, it became stale after loading from json

https://jsfiddle.net/wvnh0fxc/3/
